lib/wasi/src/syscalls/mod.rs

```diff
@@ -2318,35 +2372,25 @@ pub fn path_open<M: MemorySize>(
                     return Errno::Exist;
                 }
 
-                let write_permission = adjusted_rights.contains(Rights::FD_WRITE);
-                // append, truncate, and create all require the permission to write
-                let (append_permission, truncate_permission, create_permission) =
-                    if write_permission {
-                        (
-                            fs_flags.contains(Fdflags::APPEND),
-                            o_flags.contains(Oflags::TRUNC),
-                            o_flags.contains(Oflags::CREATE),
-                        )
-                    } else {
-                        (false, false, false)
-                    };
                 let open_options = open_options
-                    .read(true)
-                    // TODO: ensure these rights are actually valid given parent, etc.
-                    .write(write_permission)
-                    .create(create_permission)
-                    .append(append_permission)
-                    .truncate(truncate_permission);
-                open_flags |= Fd::READ;
-                if adjusted_rights.contains(Rights::FD_WRITE) {
+                    .write(minimum_rights.write)
+                    .create(minimum_rights.create)
+                    .append(minimum_rights.append)
+                    .truncate(minimum_rights.truncate);
+
+                if minimum_rights.read {
+                    open_flags |= Fd::READ;
+                }
+                if minimum_rights.write {
                     open_flags |= Fd::WRITE;
                 }
-                if o_flags.contains(Oflags::CREATE) {
+                if minimum_rights.create {
                     open_flags |= Fd::CREATE;
                 }
-                if o_flags.contains(Oflags::TRUNC) {
+                if minimum_rights.truncate {
                     open_flags |= Fd::TRUNCATE;
                 }
+
```

Comment:
no `append`? why?

Reply:
Append is the default when `WRITE` is set.

It's just the way it already was, I didn't change anything about it.